### PR TITLE
planner,executor,infoschema: add system table `table_storage_stats` (#15056)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #15056 to release-4.0

---

### What problem does this PR solve? 
we can only get the disk-usage from http API

### What is changed and how it works?
1. This pr could help us by using the table_storage_stats get the usage of a table, if a table is empty or its size < 0, its TABLE_SIZE would return 1MB.
2. Extract the get `PdRegionStats` func to helper which used by http API and this sys table.

```
mysql> select * from table_storage_stats where table_schema = 'test';
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
| TABLE_SCHEMA | TABLE_NAME | TABLE_ID | PEER_COUNT | REGION_COUNT | EMPTY_REGION_COUNT | TABLE_SIZE | TABLE_KEYS |
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
| test         | t          |      143 |          3 |            1 |                  1 |          1 |          0 |
| test         | usertable  |      343 |          3 |           97 |                  0 |       9127 |   16605517 |
| test         | t2         |      348 |          3 |            1 |                  1 |          1 |          0 |
| test         | tcs1       |      360 |          3 |           70 |                  0 |       7087 |   10204927 |
| test         | tcs2       |      362 |          3 |           70 |                  0 |       7141 |    9712408 |
| test         | tcs3       |      364 |          3 |           70 |                  0 |       7174 |    9765219 |
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
6 rows in set (0.35 sec)


mysql> select count(*) from information_schema.TABLE_STORAGE_STATS;
+----------+
| count(*) |
+----------+
|     1249 |
+----------+

mysql> select * from table_storage_stats where table_schema = 'mysql'and table_name = 'tidb';
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
| TABLE_SCHEMA | TABLE_NAME | TABLE_ID | PEER_COUNT | REGION_COUNT | EMPTY_REGION_COUNT | TABLE_SIZE | TABLE_KEYS |
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
| mysql        | tidb       |       17 |          3 |            1 |                  1 |          1 |        134 |
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
1 row in set (0.12 sec)
```
### Check List 

Tests 

 - Integration test

### Release note 

- Add system table `table_storage_stats`

### CODE REVIEW

File: infoschema/tables.go [hunk 1]
```diff
@@ -149,6 +149,8 @@ const (
 	TableTiFlashTables = "TIFLASH_TABLES"
 	// TableTiFlashSegments is the string constant of tiflash segments table.
 	TableTiFlashSegments = "TIFLASH_SEGMENTS"
+	// TableStorageStats is a table that contains all tables disk usage
```
Comment (AilinKid) [inline infoschema/tables.go:152]: This file privilege should keep it's real

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

**Review by AilinKid — COMMENTED** [review]
LGTM

**Review by tangenta — COMMENTED** [review]
LGTM

**Review by djshow832 — APPROVED** [review]
LGTM

**Review by crazycs520 — COMMENTED** [review]
LGTM

**Review by SunRunAway — APPROVED** [review]
LGTM

**Pull request merged**